File: isValidSubsequence/isValidSubsequence.py
# Time Complexity: O(N), where N represents number of elements in array
# Space Complexity: O(1)


def isValidSubsequence(array, sequence):
    seqIdx = 0

    for arrIdx in range(len(array)):
        if array[arrIdx] == sequence[seqIdx]:
            seqIdx += 1
        if seqIdx == len(sequence):
            return True

    return False

File: isValidSubsequence/isValidSubsequence.cpp
using namespace std;

bool isValidSubsequence(vector<int> array, vector<int> sequence) {
    int sequence_index = 0;

    for (auto i = array.begin(); i != array.end(); ++i) {
        if (*i == sequence[sequence_index]) {
            sequence_index++;
        }
    }

    if (sequence_index == sequence.size()) {
        return true;
    }
    else {
        return false;
    }
}
